activesupport/lib/active_support/configuration_file.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module ActiveSupport
+  class ConfigurationFile
+    INVISIBILE_WHITESPACE = /\U+A0/
+    # Reads a YAML file that might have ERB syntax, evaluates the ERB, and then
+    # parses the resulting YAML.
+    #
+    # If there are characters that will confuse YAML (Like invisible
+    # non-breaking spaces) it will warn you.
+    def initialize(content:, context: nil)
+      @content = content
+      @context = context
+    end
```

Comment:
Overall I think this is the right direction, I wanted to volley a slightly different version having seen piece:

```ruby
module ActiveSupport
  class ConfigurationFile
    # Reads a YAML file that might have ERB syntax, evaluates the ERB, and then
    # parses the resulting YAML.
    #
    # If there are characters that will confuse YAML (Like invisible
    # non-breaking spaces) it will warn you.
    def initialize(content_path)
      @content_path = content_path
      @content = read content_path
    end

    # data = ConfigurationFile.parse(@file, context: ActiveRecord::FixtureSet::RenderContext.create_subclass.new.get_binding)
    def self.parse(content_path, **options)
      new(content_path).parse(**options)
    end

    def parse(context: nil, **options)
      yaml = context ? ERB.new(@content).result(context) : ERB.new(@content).result
      YAML.safe_load(yaml, **options) || {}
    rescue Psych::SyntaxError => error
      raise Fixture::FormatError, "a YAML error occurred parsing #{@content_path}. Please note that YAML must be consistently indented using spaces. Tabs are not allowed. Please have a look at https://www.yaml.org/faq.html\nThe exact error was:\n  #{error.class}: #{error}", error.backtrace
    end

    private
      def read(content_path)
        File.read(content_path).tap do |content|
          if content.include? /\U+A0/
            warn "File contains invisible non-breaking spaces, you may want to remove those"
          end
        end
      end
  end
end
```

I don't think we want the class to know about the context, it's not central to the design, I think it should take a path that can be passed to `File.read` though, or we should rename `ConfigurationFile`.

Reply:
We can't force the user to supply a path only, because in some cases where this would be used we only have access to the content itself.

Reply:
Not internally as far as I can see, where is that?

Reply:
The major one I remember doesn't exist anymore but there's still https://github.com/rails/rails/blob/master/activerecord/lib/active_record/fixture_set/file.rb#L52-L70